src/plugins/intel_gpu/src/plugin/ops/mlir_op.cpp

```diff
@@ -38,21 +51,71 @@ void CreateMLIRSubgraphOp(ProgramBuilder& p, const std::shared_ptr<ov::op::mlir:
 
```

Comment:
We probably don't need [this synchronization](https://github.com/dchigarev/openvino/blob/4b2b653735b041c84f9656228009f78c3078360f/src/plugins/intel_gpu/src/plugin/ops/mlir_op.cpp#L41-L48) anymore since we pass the same queue to GC and submit a kernel to it. We need to figure out though on how to propagate `dependent_events` to GC in case of an out-of-order queue